unit-tests/dds/test-options.py

```diff
@@ -0,0 +1,103 @@
+    try:
+        remote.run( 'test_no_options()', timeout=5 )
+        device = dds.device( participant, participant.create_guid(), info )
+        device.run()  # If no device is available in 30 seconds, this will throw
+        test.check( device.is_running() )
+        
+        options = device.options();
+        for option in options:
+            test.unreachable() # Test no device option
+        
+        test.check_equal( device.n_streams(), 1 )
+        for stream in device.streams():
+            options = stream.options();
+            for option in options:
+                test.unreachable() # Test no stream option
+                
+        remote.run( 'close_server()', timeout=5 )
+    except:
+        test.unexpected_exception()
+    device = None
+    test.finish()
+    #
+    #############################################################################################
+    #
+    test.start( "Test device options discovery..." )
+    try:
+        remote.run( 'test_device_options_discovery()', timeout=5 )
+        device = dds.device( participant, participant.create_guid(), info )
+        device.run()  # If no device is available in 30 seconds, this will throw
+        test.check( device.is_running() )
+        
+        options = device.options();
+        test.check_equal( len( options ), 3 )
+        test.check_equal( options[0].get_value(), 1. )
+        test.check_equal( options[1].get_value(), 2. )
+        test.check_equal( options[2].get_value(), 3. )
+        
+        remote.run( 'close_server()', timeout=5 )
+    except:
+        test.unexpected_exception()
+    device = None
+    test.finish()
+    #
+    #############################################################################################
+    #
+    test.start( "Test stream options discovery..." )
+    try:
+        remote.run( 'test_stream_options_discovery()', timeout=5 )
+        device = dds.device( participant, participant.create_guid(), info )
+        device.run()  # If no device is available in 30 seconds, this will throw
+        test.check( device.is_running() )
+        test.check_equal( device.n_streams(), 1 )
+        for stream in device.streams():
+            options = stream.options();
+            test.check_equal( len( options ), 3 )
+            test.check_equal( options[0].get_value(), 1. )
+            test.check_equal( options[1].get_range().min, 0. )
+            test.check_equal( options[1].get_range().max, 123456. )
+            test.check_equal( options[1].get_range().step, 123. )
+            test.check_equal( options[1].get_range().default_value, 12. )
+            test.check_equal( options[2].get_description(), "opt3 of s1" )
+                
+        remote.run( 'close_server()', timeout=5 )
+    except:
+        test.unexpected_exception()
+    device = None
+    test.finish()
+    #
+    #############################################################################################
+
+
+participant = None
```

Comment:
I would add:
- check both device and stream options exist
- check multiple stream options exist
- check mismatch between device name (here, `Test Device`) and owner name (`device`) -- why is it not complaining?

Reply:
Added test for device and multiple stream options.

Owner name is set on receive, not sent over DDS.